api/subscriptions/serializers.py

```diff
@@ -0,0 +1,45 @@
+from rest_framework import serializers as ser
+from rest_framework.exceptions import ValidationError
+
+from api.base.serializers import JSONAPISerializer, LinksField
+
+NOTIFICATION_TYPES = {
+    'none': 'none',
+    'instant': 'email_transactional',
+    'daily': 'email_digest'
+}
+
+
+class FrequencyField(ser.Field):
+    def to_representation(self, obj):
+        user_id = self.context['request'].user.id
+        if obj.email_transactional.filter(id=user_id).exists():
+            return 'instant'
+        if obj.email_digest.filter(id=user_id).exists():
+            return 'daily'
+        return 'none'
+
+    def to_internal_value(self, data):
+        if data not in NOTIFICATION_TYPES.keys():
+            raise ValidationError('Invalid frequency "{}"'.format(data))
+        return {'notification_type': NOTIFICATION_TYPES[data]}
+
+class SubscriptionSerializer(JSONAPISerializer):
```

Comment:
Should this have a filterable_fields (type and id should be easy, the two things in attributes I'm not sure how that works)?